packages/engine/lib/memory/src/shared_memory/segment.rs

```diff
@@ -83,16 +83,18 @@ impl<'id> MemoryId<'id> {
 
     /// Clean up generated shared memory segments associated with a given `MemoryId`.
     pub fn clean_up<Id: Borrow<Uuid>>(id: Id) {
+        // We're ignoring glob errors as they shouldn't stop the whole cleanup process.
+        //
         // TODO: macOS does not store the shared memory FDs at `/dev/shm/`. Maybe it's not storing
         //   FDs at all. Find out if they are stored somewhere and remove them instead, otherwise we
         //   have to figure out a way to remove them without relying on the file-system.
-        let shm_files = glob::glob(&format!("/dev/shm/{}_*", Self::prefix(id)));
-
-        shm_files.into_iter().flatten().flatten().for_each(|path| {
-            if let Err(err) = std::fs::remove_file(&path) {
-                tracing::warn!("Could not clean up {path:?}: {err}");
-            }
-        });
+        if let Ok(shm_files) = glob::glob(&format!("/dev/shm/{}_*", Self::prefix(id))) {
+            shm_files.into_iter().flatten().for_each(|path| {
```

Comment:
As noticed on #477, we might want to remove `flatten` again altogether.

Reply:
Removed `flatten` https://github.com/hashintel/hash/pull/494/commits/5f83c33ce0f0d5a41ec302b9d6f72f6e88474e1e